### PR TITLE
Make the ConfigurationChange test stable

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Fixes https://github.com/aspnet/Logging/issues/474

Please review @BrennanConroy

### CODE REVIEW

File: test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/LogConfigurationReaderTests.cs [hunk 1]
```diff
@@ -175,11 +175,14 @@ public void ConfigurationChange()
                     Thread.Sleep(TimeSpan.FromSeconds(1));
                     settingsFileContent.AzureBlobEnabled = true;
                     settingsFileContent.AzureDriveTraceLevel = "Information";
-                    File.WriteAllText(settingsFile, JsonConvert.SerializeObject(settingsFileContent));
 
-                    var configChanged = configChangedEvent.WaitOne(DefaultTimeout);
+                    RetryHelper.Retry(() =>
+                    {
+                        // The .NET file watcher doesn't trigger all the times. We might have to retry the operation
```
Comment (BrennanConroy) [inline test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/LogConfigurationReaderTests.cs]: Is this a bug somewhere?

**Comment by victorhurdugaci** [comment]
Updated

File: test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/LogConfigurationReaderTests.cs (hunk 1 above)
Comment (victorhurdugaci) [inline test/Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Test/LogConfigurationReaderTests.cs]: Yes https://github.com/dotnet/corefx/issues/11195

**Comment by BrennanConroy** [comment]
:shipit:

**Pull request merged**